frontend/src/pages/main/transactions/components/generatePdf/GeneratePdf.tsx

```diff
@@ -0,0 +1,202 @@
+        } else {
+            setEndDays(Array.from({length: daysInMonth}, (_, i) => i + 1));
+        }
+    }
+
+    const updateReportUser = () => {
+        setReport({...report, userId: loggedUser.id});
+    }
+
+    const handleDaysChange = (event: SelectChangeEvent<number>, periodType: PeriodType) => {
+        const day = event.target.value as number;
+        if (periodType === PeriodType.START) {
+            setStartPeriod({...startPeriod, day});
+            setReport({...report, periodStart: new Date(startPeriod.year, startPeriod.month, day)});
+        } else {
+            setEndPeriod({...endPeriod, day});
+            setReport({...report, periodEnd: new Date(endPeriod.year, endPeriod.month, day)});
+        }
+    };
+
+    const handleMonthChange = (event: SelectChangeEvent<number>, periodType: PeriodType) => {
+        const month = event.target.value as number;
+        if (periodType === PeriodType.START) {
+            setStartPeriod({...startPeriod, month});
+            setReport({
+                ...report,
+                periodStart: new Date(startPeriod.year, month, startPeriod.day)
+            })
+        } else {
+            setEndPeriod({...endPeriod, month});
+            setReport({
+                ...report,
+                periodEnd: new Date(endPeriod.year, month, endPeriod.day)
+            })
+        }
+        updateDays(month, startPeriod.year, periodType);
+    };
+
+    const handleYearChange = (event: SelectChangeEvent<number>, periodType: PeriodType) => {
+        const year = event.target.value as number;
+        if (periodType === PeriodType.START) {
+            setStartPeriod({...startPeriod, year});
+            setReport({
+                ...report,
+                periodStart: new Date(year, startPeriod.month, startPeriod.day)
+            })
+        } else {
+            setEndPeriod({...endPeriod, year});
+            setReport({
+                ...report,
+                periodEnd: new Date(year, endPeriod.month, endPeriod.day)
+            })
+        }
+        updateDays(startPeriod.month, year, periodType);
+    };
+
+    const handleGenerate = async () => {
+        if (report.periodStart.getTime() > report.periodEnd.getTime()) {
+            dispatch(showNotification({show: true, type: NotificationType.ERROR, message: Notifications.PERIOD_TIME}));
+            return;
+        }
+        dispatch(showNotification({
+            show: true,
+            type: NotificationType.SUCCESS,
+            message: Notifications.SEND_GENERATE_PDF
+        }));
+        await sendReport(report);
+    }
+
+    useEffect(() => {
+        updateReportUser()
+    }, []);
+
+    return (
+        <Card className="card-background card-payment card-generate-pdf">
+            <CardContent className="card-content">
+                <Box className="balance-info">
+                    <CardTitle title={"Generate PDF report"}/>
+                    <Box className="report-date report-end-date">
```

Comment:
Looks like you can create a separate component. Then reuse it for `From` and `To` forms